pkg/deploy/skr_webhook_resources.go

```diff
@@ -0,0 +1,316 @@
+) []client.Object {
+	var genClientObjects []client.Object
+	webhookCfgObjKey := client.ObjectKey{
+		Namespace: remoteNs,
+		Name:      ResolveSKRChartResourceName(WebhookCfgAndDeploymentNameTpl, kymaObjKey),
+	}
+	svcObjKey := client.ObjectKey{
+		Namespace: remoteNs,
+		Name:      "skr-webhook",
+	}
+
+	webhookConfig := generateValidatingWebhookConfigFromWatchableConfigs(webhookCfgObjKey, svcObjKey,
+		resourcesConfig.caCert, watchableConfigs)
+	genClientObjects = append(genClientObjects, webhookConfig)
+	secretObjKey := client.ObjectKey{
+		Namespace: remoteNs,
+		Name:      ResolveSKRChartResourceName(WebhookTLSCfgNameTpl, kymaObjKey),
+	}
+	skrSecret := createSKRSecret(resourcesConfig, secretObjKey)
+	return append(genClientObjects, skrSecret)
+}
+
+func getWatchableConfigs(ctx context.Context, kcpClient client.Client) (map[string]WatchableConfig, error) {
+	watchableConfigs := map[string]WatchableConfig{}
+	watcherList := &v1alpha1.WatcherList{}
+	if err := kcpClient.List(ctx, watcherList); err != nil {
+		return nil, fmt.Errorf("error listing watcher CRs: %w", err)
+	}
+
+	watchers := watcherList.Items
+	if len(watchers) != 0 {
+		watchableConfigs = generateWatchableConfigs(watchers)
+	}
+	return watchableConfigs, nil
+}
+
+type unstructuredResourcesConfig struct {
+	contractVersion          string
+	kcpAddress               string
+	tlsWebhookServer         string
+	tlsCallback              string
+	secretName, secretResVer string
+	caCert, tlsCert, tlsKey  []byte
+	remoteNs                 string
+}
+
+//nolint:ireturn
+func configureUnstructuredResource(cfg *unstructuredResourcesConfig, resource *unstructured.Unstructured,
+) (client.Object, error) {
+	if resource.GetAPIVersion() == corev1.SchemeGroupVersion.String() && resource.GetKind() == "ConfigMap" {
+		return configureConfigMap(cfg, resource)
+	}
+	if resource.GetAPIVersion() == appsv1.SchemeGroupVersion.String() && resource.GetKind() == "Deployment" {
+		return configureDeployment(cfg, resource)
+	}
+	if resource.GetAPIVersion() == rbacV1.SchemeGroupVersion.String() && resource.GetKind() == "ClusterRoleBinding" {
+		return configureClusterRoleBinding(cfg, resource)
+	}
+	return resource, nil
+}
+
+func closeFileAndLogErr(closer io.Closer, logger logr.Logger, path string) {
+	err := closer.Close()
+	if err != nil {
+		logger.V(int(zap.DebugLevel)).Info("failed to close raw manifest file", "path", path)
+	}
+}
+
+func getRawManifestClientObjects(cfg *unstructuredResourcesConfig, remoteNs, chartPath string, logger logr.Logger,
+) ([]client.Object, error) {
+	if cfg == nil {
+		return nil, ErrExpectedNonNilConfig
+	}
+	manifestFilePath := fmt.Sprintf(rawManifestFilePathTpl, chartPath)
+	rawManifestFile, err := os.Open(manifestFilePath)
+	if err != nil {
+		return nil, err
+	}
+	defer closeFileAndLogErr(rawManifestFile, logger, rawManifestFile.Name())
+	decoder := yaml.NewYAMLOrJSONDecoder(rawManifestFile, defaultBufferSize)
```

Comment:
Why do you choose to use resources.yaml here instead of inlining the files as the proper API objects (e.g. a v1 deployment)? That way you don't have to do this yaml unstructuring and also don't have to use the runtime converter.

Reply:
I am not sure if I fully understand the question. do you suggest to remove the resources.yaml and generate the API objects in the code?

Reply:
yes exactly, that's also what you're doing with some resources already I believe

Reply:
Yes. I chose to use the yaml manifest only to avoid having a lot of code written for the sole purpose of API object generation. Do you think it's a valid concern?

Reply:
I understand the concern. I believe the correct way to solve this is then to prepare the objects in advance instead of looking them up from the file system and parsing them on every reconcile. Other than that I think you can go ahead with that approach. I just want to avoid unnecessary effort in the reconcile loop

Reply:
Makes sense. I will load them from the file system when the manager is created and configure them only when calling `Install()` 👍